sklearn/model_selection/_search.py

```diff
@@ -513,84 +518,135 @@ def _fit(self, X, y, labels, parameter_iterable):
+                                      dtype=np.int)
+
+        scorer_name = _get_scorer_name(self.scorer_)
+        results = dict()
+
+        for split_i in range(n_splits):
+            results["test_split_%d_%s" % (split_i, scorer_name)] = (
+                test_scores[:, split_i])
+
+        # Computed the (weighted) mean and std for all the candidates
+        weights = test_sample_counts if self.iid else None
+        means = np.average(test_scores, axis=1, weights=weights)
+
+        stds = np.sqrt(np.average((test_scores - means[:, np.newaxis]) ** 2,
+                                  axis=1, weights=weights))
+
+        results["test_%s_mean" % scorer_name] = means
+        results["test_%s_std" % scorer_name] = stds
+
+        ranks = rankdata(-means, method='min')
+
+        best_index = np.flatnonzero(ranks == 1)[0]
+        best_parameters = candidate_params[best_index]
+        results["test_%s_rank" % scorer_name] = ranks
+
+        for param in parameter_names:
+            # One np.MaskedArray of dtype object per parameter, mask all the
+            # places where the param is not applicable for that candidate.
+            results["param_%s" % param] = np.ma.masked_all(
+                (n_candidates,), dtype=object)
+            for i in range(n_candidates):
+                try:
+                    param_value = candidate_params[i][param]
+                    results["param_%s" % param][i] = param_value
+                except KeyError:
+                    pass
+
+        self.results_ = results
+        self.candidate_params_ = candidate_params
+        self.best_index_ = best_index
+        self.n_splits_ = n_splits
+        self.scorer_name_ = scorer_name
 
         if self.refit:
             # fit the best estimator using the entire dataset
             # clone first to work around broken estimators
             best_estimator = clone(base_estimator).set_params(
-                **best.parameters)
+                **best_parameters)
             if y is not None:
                 best_estimator.fit(X, y, **self.fit_params)
             else:
                 best_estimator.fit(X, **self.fit_params)
             self.best_estimator_ = best_estimator
         return self
 
+    def _get_candidate_scores(self, index):
+        n_splits = self.n_splits_
+        scores = np.empty((n_splits,), dtype=np.float64)
+        for split_i in range(n_splits):
+            scores[split_i] = self.results_["test_split_%s_%s"
+                                            % (split_i,
+                                               self.scorer_name_)][index]
+
+        mean = self.results_["test_%s_mean" % self.scorer_name_][index]
+        std = self.results_["test_%s_std" % self.scorer_name_][index]
+
+        return scores, mean, std
+
+    @property
+    def best_params_(self):
+        check_is_fitted(self, 'results_')
+        return self.candidate_params_[self.best_index_]
```

Comment:
This is different from the previous behaviour if `len(self.best_index_) > 1` right?